### ISSUE TITLE
Soss examples?

### ISSUE THREAD

**Issue by cscherrer** [hendersontrent/PosteriorPlots.jl#53]
Hi, I think you had said this is working for Soss, but I don't see how to use it. For example, `plot_posterior_intervals` says the first argument is a model, but using it in this way instead says it needs an `Array` or `Chains`.

I've tried instead passing it the MultiChain, and a TupleVector ( the result of `samples` which is `<: AbstractVector`) and coercing _that_ into a `Vector`. None seem to work.

Do you have some simple examples I could use as a template?

**Comment by hendersontrent** [comment]
I haven't yet tried it on all of the object types in Soss, but the following works for me:

`using Soss, Random, PosteriorPlots

model = @model X begin
    p = size(X, 2) # number of features
    α ~ Normal(0, 1) # intercept
    β ~ Normal(0, 1) |> iid(p) # coefficients
    σ ~ HalfNormal(1) # dispersion
    η = α .+ X * β # linear predictor
    μ = η # `μ = g⁻¹(η) = η`
    y ~ For(eachindex(μ)) do j
        Normal(μ[j], σ) # `Yᵢ ~ Normal(mean=μᵢ, variance=σ²)`
    end
end;

X = randn(6,2)

forward_sample = rand(model(X=X))

num_rows = 1_000
num_features = 2
X = randn(num_rows, num_features)

β_true = [2.0, -1.0]
α_true = 1.0
σ_true = 0.5

η_true = α_true .+ X * β_true
μ_true = η_true
noise = randn(num_rows) .* σ_true
y_true = μ_true .+ noise

posterior = dynamicHMC(model(X=X), (y=y_true,))

plot_posterior_intervals(posterior)`



I'll work on integrating MultiChain and TupleVector types!

**Comment by cscherrer** [comment]
I'll try that, thanks!

**Comment by hendersontrent** [comment]
Should also mention, finally had some time today and did some significant refactoring, cleaning up and adding more function flexibility. Will be much better once I finish the tests for the additions and merge it into `main`. Will work on the other Soss model types once that is merged!

**Comment by cscherrer** [comment]
Nice! Will it handle the latest version? The above sample it outdated; now you need `using MeasureTheory`, and then
```julia
julia> using SampleChainsDynamicHMC

julia> posterior = sample(model(X=X) | (y=y_true,), dynamichmc())
4000-element MultiChain with 4 chains and schema (σ = Float64, α = Float64, β = Vector{Float64})
(σ = 4.24±0.52, α = 1.0±0.13, β = [1.97±0.14, -0.956±0.13])
```

**Comment by hendersontrent** [comment]
Super clean syntax - really like the look of that updated version! `MeasureTheory.jl` is fantastic! I'll spend some time working with the latest version. `MultiChain` is first on my list to incorporate.

**Comment by hendersontrent** [comment]
So with the latest version of `PosteriorPlots` on GitHub in #58 , `MultiChain` objects are now accepted! The following should work once merged:

`using MeasureTheory
using Soss
using SampleChainsDynamicHMC
using PosteriorPlots

m = @model x begin
    α ~ Lebesgue(ℝ)
    β ~ Normal()
    σ ~ Exponential()
    y ~ For(x) do xj
        Normal(α + β * xj, σ)
    end
    return y
end

x = randn(20)

predα = predictive(m, :α)

@model (x, α) begin
        σ ~ Exponential()
        β ~ Normal()
        y ~ For(x) do xj
                Normal(α + β * xj, σ)
            end
        return y
    end

y = rand(predα(x=x,α=10.0))
post = sample(m(x=x) | (y=y,), dynamichmc())
plot_posterior_intervals(post, 0.95)`

**Comment by cscherrer** [comment]
Could we reopen this? I think it's not working properly in PosteriorPlots 0.2. Here's a gist, which should be reproducible if you load it from Pluto:
https://gist.github.com/cscherrer/e041b28251f10bbd2b7b2bac3714af90

**Comment by cscherrer** [comment]
Just a ping to make sure you see this @hendersontrent :)

**Comment by hendersontrent** [comment]
Thanks Chad! Strange, it worked for me locally when I wrote the function changes but doesn't in Pluto with your gist... I'll take a look into this. Thanks for the pickup :)

**Comment by hendersontrent** [comment]
@cscherrer this should work for your gist now as I fixed it in v0.2.1 :) Just need to add some additional plotting flexibility for the user now to make single parameter model plots look nicer.

**Comment by cscherrer** [comment]
Great! Thanks for the fix, the simple example works great now. I'll close this and let you know if other things aren't working right
